### PR TITLE
Use GFM instead of YARD code syntax in markdowns

### PR DESCRIPTION
**Pull request by kachick**
This PR is revenge of #196 😋 

https://github.com/lsegal/yard/blob/b589fa0dc0a21f3304da949dc418e0dc0032182b/docs/WhatsNew.md#github-flavoured-markdown-now-works-out-of-box-074

I didn't know YARD handles GFM as a default when using `redcarpet` in the dependency! (It is not a YARD dependency, we should specify own self)

This realize markup in YARD with GFM syntax.

screen_shot 2021-06-08 9 29 01

It works to `news.md` too!

Before
---

screen_shot 2021-06-08 9 31 05

After
---

screen_shot 2021-06-08 9 29 12

### CODE REVIEW

File: doc/text/news.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # News
 
-## 3.4.4 - 2021-06-04 {#version-3-4-4}
```
Comment (kou) [inline doc/text/news.md:3]: I want to keep the `#version-3-4-4` anchor for referencing the specified version.

File: test-unit.gemspec [hunk 2]
```diff
@@ -36,9 +36,4 @@ Gem::Specification.new do |spec|
   }
 
   spec.add_runtime_dependency("power_assert")
-  spec.add_development_dependency("bundler")
-  spec.add_development_dependency("rake")
-  spec.add_development_dependency("yard")
-  spec.add_development_dependency("redcarpet")
-  spec.add_development_dependency("packnga")
```
Comment (kachick) [inline test-unit.gemspec]: As far as I know, `add_development_dependency` is not preferable way now

* https://github.com/rubygems/bundler/pull/7222
* https://qiita.com/hirura/items/fe331672c8d41a1357e1 - ja

When specifying only in `Gemfile`, we can use `BUNDLE_WITHOUT` as https://github.com/test-unit/test-unit/pull/202/commits/325a61e3314ad5727b1b6fd6345f8b626a36ad7f#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R86-R87

**Comment by kou** [comment]
Thanks.
But I'm negative to use Redcarpet. (I had some troubles with Redcarpet before...)

Can we use https://github.com/kramdown/parser-gfm and register a custom YARD markup provider to use kramdown-parser-gfm in our `Rakefile`?

File: doc/text/news.md (hunk 1 above)
Comment (kachick) [reply doc/text/news.md:3]: `#3-4-4-2021-06-04` is the new anchor. `version-3-4-4` is the desirable for you? I can understand it👍 (Bu t I don't know hot to realize it...)

File: test-unit.gemspec (hunk 2 above)
Comment (kou) [reply test-unit.gemspec]: OK. I'm neutral for this.

**Comment by kachick** [comment]
Honestly I have tried the `kramdown-parser-gfm` in https://github.com/test-unit/test-unit/commit/709d683aed6bf22e0e8efb898cf47bb6d61e05f6 😄 
But I think using YARD official might be better.... 

>But I'm negative to use Redcarpet. (I had some troubles with Redcarpet before...)

Oh I didn't know it! 📝

File: doc/text/news.md (hunk 1 above)
Comment (kou) [reply doc/text/news.md:3]: Yes.
But it's not a strong requirement.

**Comment by kachick** [comment]
Sorry for keeping draft state in longtime, I'll close this PR for now 🙇‍♂️

**Pull request closed without merging**